integration-tests/src/integrationTest/java/pl/allegro/tech/hermes/integrationtests/TopicBlacklistTest.java

```diff
@@ -0,0 +1,99 @@
+package pl.allegro.tech.hermes.integrationtests;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.springframework.test.web.reactive.server.WebTestClient;
+import pl.allegro.tech.hermes.api.Topic;
+import pl.allegro.tech.hermes.integrationtests.setup.HermesExtension;
+import pl.allegro.tech.hermes.test.helper.message.TestMessage;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static pl.allegro.tech.hermes.api.BlacklistStatus.BLACKLISTED;
+import static pl.allegro.tech.hermes.api.BlacklistStatus.NOT_BLACKLISTED;
+import static pl.allegro.tech.hermes.test.helper.builder.TopicBuilder.topic;
+
+public class TopicBlacklistTest {
+
+    @RegisterExtension
+    public static final HermesExtension hermes = new HermesExtension();
+
+    @Test
+    public void shouldRefuseMessageOnBlacklistedTopic() {
+        // given
+        Topic topic = hermes.api().createGroupAndTopic(topic(hermes.getTestContext()).build());
+        TestMessage message = TestMessage.of("hello", "world");
+
+        // when
+        hermes.api().blacklistTopic(topic.getQualifiedName());
+        WebTestClient.ResponseSpec response = hermes.api().publish(topic.getQualifiedName(), message.body());
+
+        // then
+        response.expectStatus().isForbidden();
+    }
+
+    @Test
+    public void shouldAcceptMessageOnUnblacklistedTopic() {
+        // given
+        Topic topic = hermes.api().createGroupAndTopic(topic(hermes.getTestContext()).build());
+        TestMessage message = TestMessage.of("hello", "world");
+        hermes.api().blacklistTopic(topic.getQualifiedName());
+
+        // when
+        hermes.api().unblacklistTopic(topic.getQualifiedName());
+        WebTestClient.ResponseSpec response = hermes.api().publish(topic.getQualifiedName(), message.body());
+
+        // then
+        response.expectStatus().is2xxSuccessful();
+    }
+
+    @Test
+    public void shouldBlacklistNonExistingTopic() {
+        // expect
+        WebTestClient.ResponseSpec response = hermes.api().blacklistTopicResponse("nonExisting.topic");
+
+        // then
+        response.expectStatus().isOk();
+    }
+
+
+    @Test
+    public void shouldUnBlacklistTopic() {
+        // given
+        String topicName = hermes.getTestContext().getTestMethod().orElseThrow().getName();
+        hermes.api().blacklistTopic(topicName);
+
+        // when
+        WebTestClient.ResponseSpec response = hermes.api().unblacklistTopicResponse(topicName);
```

Comment:
in the previous test we had additional line:
`wait.untilTopicUnblacklisted("group.topic");`

Is it unnecessary?